modern/src/map/Map.js

```diff
@@ -93,6 +100,17 @@ const Map = ({ children }) => {
     };
   }, []);
 
+  useEffect(() => {
+    if (mapReady && Object.values(map.getStyle().sources).some(e => e.type === 'raster')) {
+        map.scrollZoom.setWheelZoomRate(1);
+        map.on('idle', onIdle);
+      }
+    return () => {
+      map.scrollZoom.setWheelZoomRate(1/450);
+      map.off('idle', onIdle);
+    }
```

Comment:
Can we do something like this?

```suggestion
    if (mapReady && Object.values(map.getStyle().sources).some(e => e.type === 'raster')) {
        map.scrollZoom.setWheelZoomRate(1);
        const onIdle = function () { ... };
        map.on('idle', onIdle);
        return () => {
          map.off('idle', onIdle);
        }
      } else {
         map.scrollZoom.setWheelZoomRate(1/450);
      }
```

Reply:
Yes but shouldn't `onIdle` event listener and `setWheelZoomRate` functions go together?

Reply:
Don't we only need it for fixed zoom rate?

Reply:
We use `onIdle` event listener and `setWheelZoomRate` to display map on integer zoom levels. (One for mouse wheel, one for other devices.) When we revert one back to original we should revert the other one too.

Reply:
Not sure I understand this. One is a callback. The other one is configuration.

Reply:
- Let's say map is loaded and it's zoom level is 0. When user scroll down every wheel tick will change the zoom level by 1. This is done by `setWheelZoomRate(1)` configuration. 
- Let's say this time user uses his touchpad to zoom in or out. Now, `setWheelZoomRate` is useless and user can zoom to an arbitrary zoom level. To prevent this, we add an event listener for "idle" event which fires when the map is rendered after an action. This listener checks the current zoom level, if it is not an integer forces map to zoom to the nearest integer. This way whenever a non-integer zoom level is selected, it will automatically zooms to nearest integer.

Reply:
I see. I think my solution should still work, but we can rewrite it to something like this:
```
if (mapReady && Object.values(map.getStyle().sources).some(e => e.type === 'raster')) {
        const onIdle = function () { ... };
        map.scrollZoom.setWheelZoomRate(1);
        map.on('idle', onIdle);
        return () => {
          map.scrollZoom.setWheelZoomRate(1/450);
          map.off('idle', onIdle);
        }
      }
```

Reply:
OK. Will update the PR now.